### PR TITLE
Standalone snarkwork process to submit proofs via graphql

### PR DESCRIPTION
**Pull request by deepthiskumar**
Explain your changes:
* standalone snark worker process accepts work spec in json format, accepts fee and prover public key
* Submits work to a mina node if graphql uri is passed as an argument

Explain how you tested your changes:
* ran the process with a snark work spec from devnet and sent the generated work to a devnet node

Part of solution described in https://github.com/MinaProtocol/mina/issues/16214#issuecomment-2471184848

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Review by svv232 — APPROVED** [review]

File: src/lib/mina_graphql/mina_graphql.ml [hunk 1]
```diff
@@ -811,7 +811,7 @@ module Mutations = struct
         | Ok
             ( `Broadcasted
             , Network_pool.Snark_pool.Resource_pool.Diff.Add_solved_work _
-            , _ ) ->
+            , () ) ->
```
Comment (anne-laure-s) [inline src/lib/mina_graphql/mina_graphql.ml:814]:
```suggestion
            , _ ) ->
```
It looks like this change triggers an error when building:

```
File "src/lib/mina_graphql/mina_graphql.ml", line 814, characters 14-16:
814 |             , () ) ->
                    ^^
Error: This pattern matches values of type unit
       but a pattern was expected which matches values of type
         Network_pool.Snark_pool.Resource_pool.Diff.rejected
```

Comment (svv232) [reply src/lib/mina_graphql/mina_graphql.ml]: done!

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Pull request merged**